### ISSUE TITLE
Expose test helper

### ISSUE THREAD

**Issue by mehulkar** [ember-template-lint/ember-template-lint#1218]
I want to export a lint plugin from an ember addon (the addon has other runtime stuff too), and would like a nice test-helper/framework. Currently I am bootstrapping a custom script that runs:

```js
const TemplateLinter = require('ember-template-lint');

const config = {
  plugins: ['./template-lint-plugin'],
  rules: {
    'my-rule': true
  }
};

const linter = new TemplateLinter({ config });
const testCases = [
  // my test cases  
];

const results = testCases.map(testCase => {
  const results = linter.verify({ source: testCase.template, moduleId: 'whatever.hbs' });
  const filteredFailures = results.filter(
    r => r.rule === 'my-rule'
  );

  if (testCase.shouldPass) {
    return { test: testCase.desc, passed: filteredFailures.length === 0 };
  } else {
    return { test: testCase.desc, passed: filteredFailures.length !== 0 };
  }
});

for (const result of results) {
  console.log(`${result.test}. passed: ${result.passed}`);
  if (!result.passed) {
    process.exit(1);
  }
}
```

I put this file *outside* the `tests/` directory because otherwise ember's test suite picks it up and fails because the dummy app doesn't have the plugin added (and I'm not sure I want to add it into the dummy app). 

In any case, it's not clear how to do this right.

**Comment by bmish** [comment]
I have an entirely separate npm package for my plugin called ember-template-lint-plugin-my-app.

**Comment by mehulkar** [comment]
I want/need mine to come specifically from an addon that does other stuff

**Comment by mehulkar** [comment]
I was thinking exporting https://github.com/ember-template-lint/ember-template-lint/blob/master/test/helpers/rule-test-harness.js (Even though it's just a thin wrapper of the lib version) and adding docs would be ok

**Comment by rwjblue** [comment]
The test helpers and harness are already exported in `lib/` precisely for this reason. I consider these to be public APIs, and the thin wrapper in `test/helpers/rule-test-harness.js` is all that is needed to be authored manually (it depends on your specific test framework and whatnot).

I don't know that there is anything to be done here other than properly documenting how to author new plugins (and plausibly providing a convenient blueprint or something).

**Comment by mehulkar** [comment]
@rwjblue I just revisited this yesterday, and that makes sense. I copied the jest setup in my app and the test wrapper, but am still stuck at getting the test harness to recognize the rule. The linter can't find the rule yet. I'll figure it out the wiring eventually and maybe write up a cookbook for exporting a rule from an addon.

**Comment by mehulkar** [comment]
Wrote it up here: https://mehulkar.com/blog/2020/03/ember-template-lint-rule-from-addon/

Wasn't quite sure how to add this documentation into the project itself, but feel free to grab anything that makes sense! Closing this.
